integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/aws/AwsSdxCloudStorageTests.java

```diff
@@ -0,0 +1,150 @@
+import com.sequenceiq.it.cloudbreak.cloud.v4.aws.AwsProperties;
+import com.sequenceiq.it.cloudbreak.context.Description;
+import com.sequenceiq.it.cloudbreak.context.TestContext;
+import com.sequenceiq.it.cloudbreak.dto.sdx.SdxTestDto;
+import com.sequenceiq.it.cloudbreak.exception.TestFailException;
+import com.sequenceiq.it.cloudbreak.testcase.e2e.BasicSdxTests;
+
+public class AwsSdxCloudStorageTests extends BasicSdxTests {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsSdxCloudStorageTests.class);
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Inject
+    private AwsProperties awsProperties;
+
+    @Override
+    protected void setupTest(TestContext testContext) {
+        createDefaultUser(testContext);
+        createDefaultCredential(testContext);
+        createEnvironmentWithNetworkAndFreeIPA(testContext);
+        initializeDefaultBlueprints(testContext);
+    }
+
+    @Test(dataProvider = TEST_CONTEXT)
+    @Description(
+            given = "there is a running Cloudbreak",
+            when = "a valid SDX create request with S3 Cloud Storage has been sent",
+            then = "SDX should be available AND deletable along with created S3 objects"
+    )
+    public void testSDXCanBeCreatedThenDeletedSuccessfully(TestContext testContext) throws IOException {
+        String sdx = resourcePropertyProvider().getName();
+
+        testContext
+                .given(sdx, SdxTestDto.class).withCloudStorage()
+                .when(sdxTestClient.create(), key(sdx))
+                .await(SDX_RUNNING)
+                .then((tc, testDto, client) -> {
+                    return sdxTestClient.delete().action(tc, testDto, client);
+                })
+                .await(SDX_DELETED)
+                .then((tc, testDto, client) -> {
+                    return deleteNonVersionedBucket(tc, testDto, client);
+                })
+                .validate();
+    }
+
+    private SdxTestDto deleteNonVersionedBucket(TestContext testContext, SdxTestDto sdxTestDto, SdxClient sdxClient) throws IOException {
+        AmazonS3 s3Client = createAmazonS3Client();
+        String bucketName = getBucketName();
+        AmazonS3URI amazonS3URI = new AmazonS3URI(getEUWestS3Uri());
+        String keyPrefix = getKeyPrefix(sdxTestDto);
+
+        LOGGER.info("Amazon S3 URI: {}", amazonS3URI);
+        LOGGER.info("Amazon S3 Bucket: {}", bucketName);
+        LOGGER.info("Amazon S3 Key Prefix: {}", keyPrefix);
+
+        if (s3Client.doesBucketExistV2(bucketName)) {
+            ListObjectsRequest listObjectsRequest = new ListObjectsRequest()
+                    .withBucketName(bucketName)
+                    .withPrefix(keyPrefix);
+            ObjectListing objectListing = s3Client.listObjects(listObjectsRequest);
+
+            while (true) {
+                for (S3ObjectSummary objectSummary : objectListing.getObjectSummaries()) {
+                    try {
```

Comment:
Also since we have an S3 client we could do an extra step of validation wether the cloud locations have any data inthem or not validating a much higher functionality.